src/utils/hooks.js

```diff
@@ -19,3 +19,68 @@ export function useStateWithStorage(storageKey, initialValue) {
 
 	return [value, setValue];
 }
+
+/**
+ * A custom hook that manages the importance of lists and sorts them accordingly.
+ * @param {Array<Object>} lists - An array of list objects to manage, each containing at least a `path` property.
+ * @returns {Object} An object containing:
+ *  - {Array<Object>} sortedLists - An array of lists sorted by importance and date they were pinned.
+ *  - {Function} setImportantList - A function to toggle the importance of a list identified by its `path`.
+ *  - {Function} isListImportant - A function to check if a given list is marked as important.
+ */
+export function useImportance(lists) {
+	const [sortedLists, setSortedLists] = useState([]);
+	const [importantLists, setImportantLists] = useState(
+		() => JSON.parse(localStorage.getItem('importantLists')) || [],
+	);
+
+	useEffect(() => {
+		const sortedByImportance = [...lists].sort(sortByImportance);
+		setSortedLists(sortedByImportance);
+	}, [lists, importantLists]);
+
+	const setImportantList = (listPath) => {
+		const found = isListImportant(listPath);
+		let newLists = [];
+
+		if (found) {
+			newLists = [...importantLists].filter((list) => list.path !== listPath);
+		} else {
+			newLists = [
+				{ datePinned: new Date(), path: listPath },
+				...importantLists,
+			];
+		}
+
+		setImportantLists(newLists);
+		localStorage.setItem('importantLists', JSON.stringify(newLists));
+	};
```

Comment:
Group PR review: could you add some documentation here. The code seems more on the complex side here.